### DISCUSSION TITLE
How to record mouse POV

### DISCUSSION THREAD

**Discussion by emikrish** [sn-lab/MouseGoggles#27] [Show and tell]
Hi - 

I put together a quick set of instructions if you need a POV recording of your VR scenes ...

1. SimpleScreenRecorder to record Godot scenes
- Installation: open a terminal and type: `sudo apt-get install simplescreenrecorder`
- To open the program: type in the terminal: `simplescreenrecorder`
  - Select "record the whole screen"
  - Make sure to deselect the audio recording option (if you have this selected and don't have an audio recording device, it'll fail to record)

2. Ffmpeg to edit the video
- This ffmpeg script will crop the screen recording and reorient the eyepieces. It should work on mac/windows/linux/debian.
- Navigate to the directory that has your video, then run this script (replacing with your video file name and your preferred output file name) -- if you don't have ffmpeg installed - i found it easiest to install via [homebrew](https://brew.sh/) on mac (try this [link](https://formulae.brew.sh/formula/ffmpeg#default))

`ffmpeg -i input_video.mp4 -filter_complex "[0:v]crop=240:210:(in_w-240)/2:(in_h-420)/2[top]; [0:v]crop=240:210:(in_w-240)/2:(in_h-420)/2+210,rotate=PI[bottom]; [top][bottom]vstack[combined]; [combined]transpose=1" "output_video_pov.mp4"
`

Let me know if you try it and if it works!! 🤗

**Comment by misaacson01** [comment]
Fantastic instructions, very simple and clear! 

I will also throw out an alternative idea for "recording" the screens: If you have logs of the mouse's positions (with timestamps) of a previous experiment, then you can essentially replay that experiment by altering the .gd script (or creating a version just for replays) so that it sets the position based on the log file instead of input movement. This way also lets you redefine the viewport layout so the left eye is on the left and the right eye is on the right, and whatever other changes you want to make to create a good video (like increasing the brightness or contrast). This can be a good alternative if you have any experiments where you didn't start a screen recorder.
